data_structures/binary_tree/binary_tree_maximum_path_sum.py

```diff
@@ -0,0 +1,132 @@
+"""
+A path in a binary tree is a sequence of nodes where each pair
+of adjacent nodes in the sequence has an edge connecting
+them. A node can only appear in the sequence at most once. Note
+that the path does not need to pass through the root.
+
+The path sum of a path is the sum of the node's values in the path.
+
+Given the root of a binary tree, return the maximum path sum of any non-empty path.
+
+Leetcode Reference : https://leetcode.com/problems/binary-tree-maximum-path-sum/
+"""
+
+
+class TreeNode:
```

Comment:
Can we make this a `dataclass` (see the link below)?

Reply:
@cclauss Sure, let me change it quickly !.